dbms/src/Storages/AlterCommands.cpp

```diff
@@ -718,6 +630,30 @@ void AlterCommands::validate(const StorageInMemoryMetadata & metadata, const Con
     }
 }
 
+void AlterCommands::validateDefaultExpressionForColumn(
+    const ASTPtr default_expression,
+    const String & column_name,
+    const DataTypePtr column_type,
+    const ColumnsDescription & all_columns,
+    const Context & context) const
+{
+
+    try
+    {
+        String tmp_column_name = "__tmp" + column_name;
+        auto copy_expression = default_expression->clone();
+        auto default_with_cast = makeASTFunction("CAST", copy_expression, std::make_shared<ASTLiteral>(column_type->getName()));
+        auto query_with_alias = setAlias(default_with_cast, tmp_column_name);
+        auto syntax_result = SyntaxAnalyzer(context).analyze(query_with_alias, all_columns.getAll());
+        ExpressionAnalyzer(query_with_alias, syntax_result, context).getActions(true);
+    }
+    catch (Exception & ex)
+    {
+        ex.addMessage("default expression and column type are incompatible. Cannot alter column '" + column_name + "'");
+        throw(ex);
```

Comment:
Just `throw`. It's important to keep full stack trace.

There is a difference between rethrowing and throwing a new exception.